### ISSUE TITLE
[Bug] Some patients cured are not counted in money earnings

### ISSUE THREAD

**Issue by carlosrd** [CorsixTH/CorsixTH#2064]
#### Describe the issue
I don't know exactly in what conditions, but it happens in any level at any random moment, that sometimes, you cure people in the hospital but this is not reflected in money earnings.

This is very noticeable when you are out of money and cure a patient. You can listen the "celebration voices" but money is not incremented.

#### Steps to Reproduce
1. Start a game and pay attention for earnings in some room (Pharmacy, Operation Theatre, etc.). 
2. Check if all the cures in that room are counted in money earnings.
3. Note that some of them are ignored in money earnings

I've attached a couple of screenshots where you can see this effect, but it can happen at any moment and is recurrently

#### Expected Behavior
All patients cured are reflected in money earnings

#### System Information
**CorsixTH Version:** 0.65

**Operating System:** Windows 10 21H1 Version (Compilation 19043.1237)

**Theme Hospital Version:** CD

### Additional Info
In this screenshot you can see patient recently opered in Operation Theatre by $1120 at 9th-10th Jan (Ene):
![patient_cured](https://user-images.githubusercontent.com/2896275/134824558-e7b6d9bf-8b87-46a1-a11a-a22988ed959d.png)

But money earnings are not incremented and those $1120 are not reflected in hospital contability:
![hospital_contability](https://user-images.githubusercontent.com/2896275/134824639-ce8b9b08-74df-4eec-8880-9019c7eab647.png)
If you check incomes (_Dentro_) column for 9-15 Jan, there are only:
* _Tratamientos: Medicina General_ (patients paying attention at GP offices)
* _Tratamientos: Diag. General_ (patients paying at General Diagnosis)
* _Ingresos: Maquina de bebidas_ (patients buying drinks)
* _Cura: Ulcera Gástrica_ (cure at Pharmacy for Gastric Ulcer)

**Comment by lewri** [comment]
I'm suspecting this one is normal.
When a patient pays for treatment they may opt to use insurance. As a result, insurance payments get delayed in when they are paid to you.
You can see this usually in the Bank Manager screen.

**Comment by lewri** [comment]
Slightly off topic but I've pondered on the idea of having a different colour dollar sign when a patient uses insurance (e.g. blue). Though it would deviate from original a bit.

**Comment by Alberth289346** [comment]
Arguably, the "Saldo" column is wrong, the hospital doesn't have the money at that time.
If that is changed, at the end of the month there should be an entry for the received money then.

**Comment by MRMIdAS** [comment]
Yeah, this is standard behaviour, think of the money as coming off your bills at the end of the month, so if you cure a bunch of insurance patients, your wage bill at the end of that month is less.

**Comment by lewri** [comment]
> Arguably, the "Saldo" column is wrong, the hospital doesn't have the money at that time. If that is changed, at the end of the month there should be an entry for the received money then.

I believe that refers to this item at the top: "seguro: el seguro fiel" -- seguro meaning insurance.

**Comment by lewri** [comment]
Information about insurance payments should be added to the game guide as suggested by @Alberth289346 
Then this can be closed.

**Comment by SilPho** [comment]
This does seem to be included in the Game Guide already
https://github.com/CorsixTH/CorsixTH/wiki/Game-Guide#at-the-bank

It's not super obvious that this means the floating payment icons aren't immediately added to your account, I agree with Lewri that another colour might be helpful here, but as implemented I'd say the wiki covers it well enough.

**Comment by lewri** [comment]
> This does seem to be included in the Game Guide already https://github.com/CorsixTH/CorsixTH/wiki/Game-Guide#at-the-bank
> 
> It's not super obvious that this means the floating payment icons aren't immediately added to your account, I agree with Lewri that another colour might be helpful here, but as implemented I'd say the wiki covers it well enough.

Yep
```
The Insurance Buttons on the right of the screen show you how your Patients’ Insurance companies are paying you. Some Patients pay you in cash but others let you bill their Insurance companies. Some Insurance companies take more time to pay than others do (not implemented, they all take about 3 months). Left-click on their graphs to find out how efficiently your cash is flowing in. (not implemented)
```

I would also probably prefer clearing this up to make it a little more obvious. Maybe drawing attention to the balance sheet to further demonstrate the matter.

**Comment by ARGAMX** [comment]
This problem is related to the fact that it’s not obvious that the patient was treated under the insurance policy as well #1375
